pkg/controller/api/radapp.io/v1alpha3/deploymentresource_types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DeploymentResourceSpec defines the desired state of DeploymentResource
+type DeploymentResourceSpec struct {
+	// Id is the resource Id.
+	Id string `json:"id,omitempty"`
+
+	// ProviderConfig specifies the scope for resources
+	ProviderConfig string `json:"providerConfig,omitempty"`
+
+	// RootFileName is the name of the Bicep file that
+	// `bicep build` is run on.
+	RootFileName string `json:"rootFileName,omitempty"`
+}
+
+// DeploymentResourceStatus defines the observed state of DeploymentResource
+type DeploymentResourceStatus struct {
+	// Id is the resource Id.
+	Id string `json:"id,omitempty"`
+
+	// ProviderConfig specifies the scope for resources
+	ProviderConfig string `json:"providerConfig,omitempty"`
+
+	// RootFileName is the name of the Bicep file that
+	// `bicep build` is run on.
+	RootFileName string `json:"rootFileName,omitempty"`
```

Comment:
It looks like all of the fields from the spec also appear in the status. What does this do for us?